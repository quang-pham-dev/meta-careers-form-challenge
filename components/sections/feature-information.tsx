import { FeatureDescription } from '@/components/sections/feature-description'
import { FEATURES_INFORMATION } from '@/constants'

export function FeatureInformation() {
  return (
    <>
      <h1 className="text-4xl font-bold mb-4">Sign up for Career Profile</h1>
      <p className="text-lg mb-6">
        Create a Career Profile to keep track of your applications, prepare for
        the interview process and more.
      </p>
      <p className="text-lg text-gray-600 mb-2">Additional Features</p>
      <div className="w-36 h-0.5 bg-gray-300 mb-4"></div>

      <div className="space-y-4">
        {FEATURES_INFORMATION.map(feature => {
          const { title, description } = feature
          return (
            <FeatureDescription
              key={title}
              title={title}
              description={description}
            />
          )
        })}
      </div>
    </>
  )
}
