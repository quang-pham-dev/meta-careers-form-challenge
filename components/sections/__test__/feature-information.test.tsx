import React from 'react'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { vi } from 'vitest'

import { FeatureInformation } from '../feature-information'

// Mock the constants module
vi.mock('@/constants', () => ({
  FEATURES_INFORMATION: [
    { title: 'Feature 1', description: 'Description 1' },
    { title: 'Feature 2', description: 'Description 2' },
  ],
}))

vi.mock('@/components/sections/feature-description', () => ({
  FeatureDescription: ({
    title,
    description,
  }: {
    title: string
    description: string
  }) => (
    <div data-testid="feature-description">
      <h3>{title}</h3>
      <p>{description}</p>
    </div>
  ),
}))

describe('FeatureInformation', () => {
  it('renders the component correctly', () => {
    render(<FeatureInformation />)

    // Check if the main title is rendered
    expect(screen.getByText('Sign up for Career Profile')).toBeInTheDocument()

    // Check if the description paragraph is rendered
    expect(
      screen.getByText(
        /Create a Career Profile to keep track of your applications/,
      ),
    ).toBeInTheDocument()

    // Check if "Additional Features" text is rendered
    expect(screen.getByText('Additional Features')).toBeInTheDocument()

    // Check if all feature descriptions are rendered
    const featureDescriptions = screen.getAllByTestId('feature-description')
    expect(featureDescriptions).toHaveLength(2) // Use the mocked length

    // Check if each feature title and description is rendered
    expect(screen.getByText('Feature 1')).toBeInTheDocument()
    expect(screen.getByText('Description 1')).toBeInTheDocument()
    expect(screen.getByText('Feature 2')).toBeInTheDocument()
    expect(screen.getByText('Description 2')).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<FeatureInformation />)
    expect(asFragment()).toMatchSnapshot()
  })
})
