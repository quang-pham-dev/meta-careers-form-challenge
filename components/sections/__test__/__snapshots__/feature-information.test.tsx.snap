// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeatureInformation > matches snapshot 1`] = `
<DocumentFragment>
  <h1
    class="text-4xl font-bold mb-4"
  >
    Sign up for Career Profile
  </h1>
  <p
    class="text-lg mb-6"
  >
    Create a Career Profile to keep track of your applications, prepare for the interview process and more.
  </p>
  <p
    class="text-lg text-gray-600 mb-2"
  >
    Additional Features
  </p>
  <div
    class="w-36 h-0.5 bg-gray-300 mb-4"
  />
  <div
    class="space-y-4"
  >
    <div
      data-testid="feature-description"
    >
      <h3>
        Feature 1
      </h3>
      <p>
        Description 1
      </p>
    </div>
    <div
      data-testid="feature-description"
    >
      <h3>
        Feature 2
      </h3>
      <p>
        Description 2
      </p>
    </div>
  </div>
</DocumentFragment>
`;
